common/djangoapps/student/tests/test_parental_controls.py

```diff
@@ -0,0 +1,87 @@
+        """Verify the behavior for users with no specified year of birth."""
+        self.assertTrue(self.profile.requires_parental_consent())
+        self.assertTrue(self.profile.requires_parental_consent(default_requires_consent=True))
+        self.assertFalse(self.profile.requires_parental_consent(default_requires_consent=False))
+
+    @override_settings(PARENTAL_CONSENT_AGE_LIMIT=None)
+    def test_no_parental_controls(self):
+        """Verify the behavior for all users when parental controls are not enabled."""
+        self.assertFalse(self.profile.requires_parental_consent())
+        self.assertFalse(self.profile.requires_parental_consent(default_requires_consent=True))
+        self.assertFalse(self.profile.requires_parental_consent(default_requires_consent=False))
+
+        # Verify that even a child does not require parental consent
+        current_year = datetime.datetime.now().year
+        self.set_year_of_birth(current_year - 10)
+        self.assertFalse(self.profile.requires_parental_consent())
+
+    def test_adult_user(self):
+        """Verify the behavior for an adult."""
+        current_year = datetime.datetime.now().year
+        self.set_year_of_birth(current_year - 20)
+        self.assertFalse(self.profile.requires_parental_consent())
+        self.assertTrue(self.profile.requires_parental_consent(age_limit=21))
+
+    def test_child_user(self):
+        """Verify the behavior for a child."""
+        current_year = datetime.datetime.now().year
+
+        # Verify for a child born 13 years agp
+        self.set_year_of_birth(current_year - 13)
+        self.assertTrue(self.profile.requires_parental_consent())
+        self.assertTrue(self.profile.requires_parental_consent(date=datetime.date(current_year, 12, 31)))
+        self.assertFalse(self.profile.requires_parental_consent(date=datetime.date(current_year + 1, 1, 1)))
+
+        # Verify for a child born 14 years ago
+        self.set_year_of_birth(current_year - 14)
+        self.assertFalse(self.profile.requires_parental_consent())
+        self.assertFalse(self.profile.requires_parental_consent(date=datetime.date(current_year, 1, 1)))
+
+    def test_profile_image(self):
+        """Verify that a profile's image obeys parental controls."""
+
+        # Verify that an image cannot be set for a user with no year of birth set
+        self.profile.has_profile_image = True
+        self.profile.save()
+        self.assertFalse(self.profile.has_profile_image)
+
+        # Verify that an image can be set for an adult user
+        current_year = datetime.datetime.now().year
+        self.set_year_of_birth(current_year - 20)
+        self.profile.has_profile_image = True
+        self.profile.save()
+        self.assertTrue(self.profile.has_profile_image)
+
+        # verify that a user's profile image is removed when they switch to requiring parental controls
+        self.set_year_of_birth(current_year - 10)
+        self.profile.has_profile_image = True
```

Comment:
Nit: the test seems clearer to me if you delete line 85 (since the user had a profile image before you changed the year_of_birth).